src/features/posts/posts.ts

```diff
@@ -0,0 +1,53 @@
+import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
+import { Post } from '../../types/Post';
+import { getUserPosts } from '../../api/posts';
+
+export const fetchUserPosts = createAsyncThunk(
+  'posts/fetch',
+  (userId: number) => getUserPosts(userId),
+);
+
+type InitialState = {
+  selectedPost: Post | null;
+  posts: Post[];
+  loaded: boolean;
+  hasError: boolean;
+};
+
+const initialState: InitialState = {
+  selectedPost: null,
+  posts: [],
+  loaded: true,
```

Comment:
The initial state for `loaded` should be `false` to correctly represent that data loading has not started. This will prevent any UI components from mistakenly assuming data is ready before any fetch operation.